parlai/core/params.py

```diff
@@ -1043,10 +1052,10 @@ def _kwargs_to_str_args(self, **kwargs):
                 string_args.append(last_option_string)
```

Comment:
This chunk (and its mirror below) needs to become:

```
if isinstance(action, ,argparse._StoreActionTrue):
    if bool(value):
        string_args.append(last_option_string)
```
That should fix the advanced args issue